keps/sig-api-machinery/5073-declarative-validation-with-validation-gen/README.md

```diff
@@ -0,0 +1,1285 @@
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Beta & GA:
+
+A rollout can fail by being too strict with updates. If the ported declarative validations are stricter than native validations then workloads may not be able to execute their update operations.
+
+A rollback can fail if our declarative validations are too loose. Workloads wont be able to update objects with an invalid field until the object is corrected.
+
+###### What specific metrics should inform a rollback?
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+        This section must be completed when targeting beta to a release.
+
+        For GA, this section is required: approvers should be able to confirm the previous answers based on experience in the field.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+```bash
+    $ kubectl get --raw /metrics | grep DeclarativeValidation
+    # look for if the featuregate is enabled from /metrics
+kubernetes_feature_enabled{name="DeclarativeValidation",stage=""} 1
+```
+
+###### How can someone using this feature know that it is working for their instance?
+
+This is not a user controllable feature
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+We expect to hit the same performance numbers for API requests as we have with just handwritten validation.  This will be benchmarked by looking at `apiserver_request_duration_seconds`.  In the case that declarative validation has meaningful performance impact, we believe there are performance improvements for validation generally that can be done to mitigate this.  For more information see the section - "Risk: Added latency to API request handling."
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+4. Metrics
+    *   Metric name: `apiserver_request_duration_seconds`
+    *   Metric name: `declarative_validation_mismatch`
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+A useful metric for improving observability of this feature exists - apiserver_request_duration_seconds.  This metric allows us to compare how long requests take w/ and w/o `DeclarativeValidation` set which allows us to benchmark our implementation and understand any performance implications and allow us to mitigate them.  
+
+A missing metric that would potentially be useful would be more granular metrics of duration across of each part of the apiserver handler chain, in our case something like: `apiserver_request_validation_duration_seconds` would be even more helpful in comparing the performance of hand-written validaton vs declarative validation as the project progresses.
```

Comment:
delta on `apiserver_request_duration_seconds` is likely good enough, but if we need it ok.